tfhe/examples/utilities/params_to_file.rs

```diff
@@ -0,0 +1,172 @@
+/// Xe=ND.DiscreteGaussian(56139.60810663548), tag='test_lattice_estimator')
+pub fn format_glwe_parameters_to_lattice_estimator<T: Spam + NamedParam>(
+    param: &T,
+    log_ciphertext_modulus: usize,
+) -> String {
+    let modular_std_dev = log_ciphertext_modulus as f64 + param.glwe_modular_std_dev().0.log2();
+
+    format!(
+        "{}_GLWE = LWE.Parameters(\n n = {},\n q = {},\n Xs=ND.UniformMod(2), \n Xe=ND.DiscreteGaussian({}),\n tag='{}_glwe' \n)\n\n",
+        param.name(), param.glwe_dimension().0*param.polynomial_size().0, (1u128<<log_ciphertext_modulus as u128), 2.0_f64.powf(modular_std_dev), param.name())
+}
+
+fn write_file(file: &mut File, filename: &Path, line: impl Into<String>) {
+    let error_message = format!("unable to write file {}", filename.to_str().unwrap());
+    file.write_all(line.into().as_bytes())
+        .expect(&error_message);
+}
+
+fn write_all_params_in_file<T: Spam + Copy + NamedParam>(
+    filename: &str,
+    params: &[T],
+    log_ciphertext_modulus: usize,
+) {
+    let path = Path::new(filename);
+    File::create(path).expect("create results file failed");
+    let mut file = OpenOptions::new()
+        .append(true)
+        .open(path)
+        .expect("cannot open parsed results file");
+
+    for params in params.iter() {
+        write_file(
+            &mut file,
+            path,
+            format_lwe_parameters_to_lattice_estimator(params, log_ciphertext_modulus),
+        );
+        write_file(
+            &mut file,
+            path,
+            format_glwe_parameters_to_lattice_estimator(params, log_ciphertext_modulus),
+        );
+    }
+    write_file(&mut file, path, "all_params = [\n");
+    for params in params.iter() {
+        let param_lwe_name = format!("{}_LWE,", params.name());
+        write_file(&mut file, path, param_lwe_name);
+
+        let param_glwe_name = format!("{}_GLWE,", params.name());
+        write_file(&mut file, path, param_glwe_name);
+    }
+    write_file(&mut file, path, "\n]\n");
+}
+
+fn main() {
+    let work_dir = std::env::current_dir().unwrap();
+    let mut new_work_dir = work_dir;
+    new_work_dir.push("ci");
+    std::env::set_current_dir(new_work_dir).unwrap();
+
+    write_all_params_in_file(
+        "boolean_parameters_lattice_estimator.sage",
+        &VEC_BOOLEAN_PARAM,
+        32,
+    );
+
+    let classic_pbs = ALL_PARAMETER_VEC
+        .iter()
+        .map(|p| ShortintParameterSet::from(*p))
+        .collect::<Vec<_>>();
+    write_all_params_in_file(
+        "shortint_classic_parameters_lattice_estimator.sage",
+        &classic_pbs,
+        64,
+    );
+
+    let multi_bit_pbs = ALL_MULTI_BIT_PARAMETER_VEC
+        .iter()
+        .map(|p| ShortintParameterSet::from(*p))
+        .collect::<Vec<_>>();
+    write_all_params_in_file(
+        "shortint_multi_bit_parameters_lattice_estimator.sage",
+        &multi_bit_pbs,
+        64,
+    );
+
+    // TODO perform this gathering later
```

Comment:
todo now or later ?

Reply:
Later. Since the margin for the optimizer is too tight for now.

Reply:
Okay